### ISSUE TITLE
Putting TestIDs in Content Description for Appium

### ISSUE THREAD

**Issue by talenfisher** [wix-playground/ahbottomnavigation#5]
Hello!  My team uses Appium for e2e tests, which does accessibility label lookups to find elements.  On Android, this translates to looking up elements by their content description.  

We've been unable to grab react-native-navigation tabs by their testID, because the content description contains other accessibility text:

https://github.com/wix-playground/ahbottomnavigation/blob/b0db704605e34707bd4b425b92e522667695af24/ahbottomnavigation/src/main/java/com/aurelhubert/ahbottomnavigation/AHBottomNavigation.java#L660-L670

Would you accept a PR setting the content description of the tabs to their testID instead of what they are now?

**Comment by swabbass** [comment]
> Hello! My team uses Appium for e2e tests, which does accessibility label lookups to find elements. On Android, this translates to looking up elements by their content description.
> 
> We've been unable to grab react-native-navigation tabs by their testID, because the content description contains other accessibility text:
> 
> https://github.com/wix-playground/ahbottomnavigation/blob/b0db704605e34707bd4b425b92e522667695af24/ahbottomnavigation/src/main/java/com/aurelhubert/ahbottomnavigation/AHBottomNavigation.java#L660-L670
> 
> Would you accept a PR setting the content description of the tabs to their testID instead of what they are now?

Hey @talenfisher, 

Today we are using tags for test ids, Appuim can use tags when using Espresso only, but, in order to use `contentDescription` you may consider apps that use `contentDescription` for its intended usage, and prevent clash with values of such attribute. 

My suggestion is to plan a way to switch `contentDescription` based on the environment, test or production, which can be even generic for each view. wdyt?

**Comment by talenfisher** [comment]
> My suggestion is to plan a way to switch `contentDescription` based on the environment, test or production

I like that - good call.

> which can be even generic for each view. wdyt?

By this, do you mean relying on a property passed down through react-native-navigation to determine whether contentDescription or viewtag should be used for the testID?  I'm not sure I'd want to do this for each individual tab.  However, if it's just a single property that decides this for every tab in the bottom navigation, I think that is definitely preferable.

**Comment by swabbass** [comment]
yup, on android for each view you can call `setContentDescription` and that simulates the accessibility id, but, I found it troublesome, see this old [discussion](https://groups.google.com/g/appium-discuss/c/Uh2QyCHMyI0), it might give you a basic idea of what issues you may face, especially using `TextViews`, `EditTexts` and other non-image views.
